### ISSUE TITLE
MNN在联发科天玑9500设备上开启SME2存在输出异常

### ISSUE THREAD

**Issue by mistletoe1024** [alibaba/MNN#4828]
MNN 在 天玑9500设备 上启用 SME2 打开的量化/FP16 计算路径产生了错误数值；
最终让 logits 退化并连续选择同一 token。
并且开关 SME2 是唯一变量，可以确认问题边界；

模型权重：qwe3.5-0.8b(llmexport.py转换)
设备：vivo x300 联发科天玑9500

现象：
1、原始 MNN、原始模型、原始 `precision=low`/4 线程在天玑 9500 上 5 次有 1 次失败，最长连续输出 506 个 `!`；
2、将采样改为 greedy 后仍曾出现重复，说明随机 sampler 不是根因；调 repetition penalty 只能掩盖症状，不能修正错误 logits；
3、相同版本源码、相同旧模型，仅改为 `-DMNN_SME2=OFF`，3/3 未复现。
4、后续代码包含 Qwen3.5 LinearAttention FP16 累积精度修复，但带 SME2 的后续构建仍能复现；
5、截至9.2的最新 `master@d56423df`，使用同一快照重新转换模型后，SME2 开启 5 次有 2 次失败，分别最长连续输出 486 和 221 个 `8`；这与现场的“8888”症状完全一致；
6、同一份源码、同一份最新模型、同一配置，仅关闭 SME2 后连续 10/10 正常，未发现连续 20 个以上相同字符，也未出现 `!!!!!!!!!!!!!!!!!!!!` 或 `88888888888888888888`

**Comment by v0jiuqi** [comment]
1. 你的vivo x300系统有没有升级到最新了，如果是老的系统软件版本，sme2就会出错。不是MNN的问题。
2. 0.8B 模型在导出的时候试试加上 --hqq，可以提升推理精度。
3. 如果还是有问题，拉最新的master再编译试一次
4. 如果还有问题，就把你复现的prompt发我。

**Comment by mistletoe1024** [comment]
已确认是系统问题，更新后开启SME2正常，感谢回复！
